File: Sample.BlazorWasm.TextEditor/wwwroot/src/Common/Interop/JSEvent.ts
namespace Sample.BlazorWasm.TextEditor {
    /**
     * 
     */
    export class JSEvent {
        /**
         *
         */
        public constructor(
            private readonly _objRef: DotNetObjectReference) {
        }

        /**
         *
         */
        public raiseAction(eventName: string, arg: any): void {
            this._objRef.invokeMethod("CallbackAction", eventName, arg);
        }

        /**
         *
         */
        public raiseFunc(eventName: string, arg: any): any {
            return this._objRef.invokeMethod("CallbackFunc", eventName, arg);
        }
    }
}

File: Sample.BlazorWasm.TextEditor/Common/Interop/JSEvent.cs
using System;
using System.Collections.Generic;
using System.Text.Json;
using Crash.Core.Collections;
using Microsoft.JSInterop;

namespace Sample.BlazorWasm.TextEditor.Common.Interop
{
    public sealed class JSEvent
    {
        private readonly Dictionary<string, Delegate> _listenerMap = new Dictionary<string, Delegate>();

        public void RegisterActionListener(string eventName, Action<JsonElement> callback)
        {
            _listenerMap.Add(eventName, callback);
        }

        public void RegisterFuncListener(string eventName, Func<JsonElement, object> callback)
        {
            _listenerMap.Add(eventName, callback);
        }

        /// <summary>
        /// 
        /// </summary>
        [JSInvokable]
        public void CallbackAction(string eventName, JsonElement jsonElem)
        {
            _listenerMap.GetValueOrNull(eventName)?.DynamicInvoke(jsonElem);
        }

        /// <summary>
        /// 
        /// </summary>
        [JSInvokable]
        public object? CallbackFunc(string eventName, JsonElement jsonElem)
        {
            return _listenerMap.GetValueOrNull(eventName)?.DynamicInvoke(jsonElem);
        }
    }
}
